### PR TITLE
[csharp,android] Add Xamarin Android support

### LINKED ISSUE #276: Will this work with Xamarin?
Can this be used with Xamarin iOS / Android?

### PR DESCRIPTION
**Pull request by rmarinho**
Adds a Xamarin Android support for Yoga and respective tests. Closes #276 with this PR all Xamarin platforms will be covered

Need to figure a better strategy for the build names for buck with @splhack .

It's failing 3 tests related with YogaNode and the GC :) classic! 

screenshot_1484353863

```
Facebook.Yoga.Android.Tests.dll : 660.87 ms
 : 678.788 ms
Tests run: 130, Passed: 127, Failed: 3, Skipped: 0, Inconclusive: 0
```

### CODE REVIEW

File: csharp/Android/Facebook.Yoga.Android/Facebook.Yoga.Android.csproj [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	</ItemGroup>
+	<ItemGroup>
+		<None Include="CustomBuildAction.targets" />
+	</ItemGroup>
+	<ItemGroup>
+	  <Folder Include="lib\x86\" />
+	  <Folder Include="lib\armeabi-v7a\" />
+	</ItemGroup>
+	<ItemGroup>
+	  <EmbeddedNativeLibrary Include="lib\x86\libyoga.so">
```
Comment (rmarinho) [inline csharp/Android/Facebook.Yoga.Android/Facebook.Yoga.Android.csproj]: This is the only important bit. `EmbeddedNativeLibrary`

File: csharp/Android/Facebook.Yoga.Android/CustomBuildAction.targets [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+	<PropertyGroup>
+		<CompileDependsOn>CopyInNativeLib;$(CompileDependsOn)</CompileDependsOn>
```
Comment (splhack) [inline csharp/Android/Facebook.Yoga.Android/CustomBuildAction.targets]: <?xml version="1.0" encoding="UTF-8"?>
    <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    	<PropertyGroup>
            <CompileDependsOn>NativeLibrary;$(CompileDependsOn)</CompileDependsOn>
        </PropertyGroup>
        <Target Name="NativeLibrary" Outputs="$(ProjectDir)libyoga.a">
            <Exec WorkingDirectory="$(ProjectDir)" Command="$(ProjectDir)../../Mac/buck-build.sh //yoganet#android-armv7,shared" />
            <Exec WorkingDirectory="$(ProjectDir)" Command="$(ProjectDir)../../Mac/buck-build.sh //yoganet#android-x86,shared" />
            <Copy SourceFiles="$(ProjectDir)..\..\..\buck-out\gen\csharp\yoganet#android-armv7,shared\libyoga.so" DestinationFiles="$(ProjectDir)/lib/armeabi-v7a/libyoga.so" SkipUnchangedFiles="true" />
            <Copy SourceFiles="$(ProjectDir)..\..\..\buck-out\gen\csharp\yoganet#android-x86,shared\libyoga.so" DestinationFiles="$(ProjectDir)/lib/x86/libyoga.so" SkipUnchangedFiles="true" />
        </Target>
    </Project>

and I've realized we're using `/` and `\` for path in Mac and iOS as well...

**Comment by emilsjolander** [comment]
@splhack is this good to go?

**Comment by splhack** [comment]
My comment is here https://github.com/facebook/yoga/pull/340/files#r96103686
(Maybe we'll be able to remove all CustomBuildAction.targets if we could release NuGet binary libyoga libraries)

**Comment by splhack** [comment]
@rmarinho can you remove `csharp/Android/Facebook.Yoga.Android/lib/armeabi-v7a/libyoga.so`?

**Comment by splhack** [comment]
never mind. I'll remove it in merge commit.

**Pull request closed without merging**